File: util/card.py
class Card:
    suits = ["Hearts", "Diamonds", "Spades", "Clubs"]
    ranks = [str(i) for i in range(2,11)] + ["Jack", "Queen", "King", "Ace"]
    
    def __init__(self, suit, rank):
        self.suit = suit
        self.rank = rank
        self.name = self.ranks[rank] + " of " + self.suits[suit]

    def __str__(self):
        return self.name

    def __repr__(self):
        return self.name

File: game_server/src/objects/card.ts

const suits: string[] = ["Hearts", "Diamonds", "Spades", "Clubs"]
const ranks: string[]  = Array.from(Array(9).keys()).map(i => String(i + 2)).concat(["Jack", "Queen", "King", "Ace"])

export class Card {
    suit: number;
    rank: number;
    name: string;

    constructor(suit: number, rank: number) {
        if (suit < 0 || suit > 3 || rank < 0 || rank > 12) {
            throw new Error("Invalid card");
        }
        this.suit = suit;
        this.rank = rank;

        this.name = ranks[rank] + " of " + suits[suit];
    }

    toFirebaseObject() {
        return {
            suit: this.suit,
            rank: this.rank,
            name: this.name
        }
    }

    equals(other: Card) {
        return this.suit === other.suit && this.rank === other.rank;
    }

    toString() {
        return this.name;
    }
}
